server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -2,64 +2,40 @@ package cached_mysql
 
 import (
 	"context"
-	"encoding/json"
-	"fmt"
+	"sync"
 	"time"
 
-	"github.com/fleetdm/fleet/v4/server/datastore/redis"
 	"github.com/fleetdm/fleet/v4/server/fleet"
-	redigo "github.com/gomodule/redigo/redis"
 )
 
 type cachedMysql struct {
 	fleet.Datastore
 
-	redisPool fleet.RedisPool
+	mu        sync.Mutex
+	ac        *fleet.AppConfig
+	acLastErr error
 }
 
-const (
-	CacheKeyAppConfig = "cache:AppConfig"
-)
+func New(ctx context.Context, ds fleet.Datastore) fleet.Datastore {
+	cds := &cachedMysql{Datastore: ds}
+	go cds.refresher(ctx)
 
-func New(ds fleet.Datastore, redisPool fleet.RedisPool) fleet.Datastore {
-	return &cachedMysql{
-		Datastore: ds,
-		redisPool: redisPool,
-	}
+	return cds
 }
 
-func (ds *cachedMysql) storeInRedis(key string, v interface{}) error {
-	conn := redis.ConfigureDoer(ds.redisPool, ds.redisPool.Get())
-	defer conn.Close()
-
-	b, err := json.Marshal(v)
-	if err != nil {
-		return fmt.Errorf("marshaling object to cache in redis: %w", err)
-	}
-
-	if _, err := conn.Do("SET", key, b, "EX", (24 * time.Hour).Seconds()); err != nil {
-		return fmt.Errorf("caching object in redis: %w", err)
-	}
-
-	return nil
-}
-
-func (ds *cachedMysql) getFromRedis(key string, v interface{}) error {
-	conn := redis.ReadOnlyConn(ds.redisPool,
-		redis.ConfigureDoer(ds.redisPool, ds.redisPool.Get()))
-	defer conn.Close()
-
-	data, err := redigo.Bytes(conn.Do("GET", key))
-	if err != nil {
-		return fmt.Errorf("getting value from cache: %w", err)
-	}
-
-	err = json.Unmarshal(data, v)
-	if err != nil {
-		return fmt.Errorf("unmarshaling object from cache: %w", err)
+func (ds *cachedMysql) refresher(ctx context.Context) {
```

Comment:
I worry that this function could get pretty messy if we start caching more data.

I'm not sure this is simpler, and I'm not demanding that we do it this way, but my idea:

Keep a "last updated" time for the cache value and update it "on demand" when a call to that method sees that the value is "out of date" (eg. more than 1s old).

This way the update logic could be kept in each method rather than centrally here. It would also potentially reduce the total number of calls to the backing store, but not if there is significant load on the system (because the refresh would always happen immediately after expiration).